### PR TITLE
added rbac roles and binding

### PR DESCRIPTION
**Pull request by leahnp**
Added RBAC and cleaned up the daemon-set

### CODE REVIEW

File: fluentd-daemonset.yaml [hunk 1]
```diff
@@ -1,23 +1,77 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: fluentd-daemon
+  namespace: kube-system
```
Comment (alejandroEsc) [inline fluentd-daemonset.yaml]: just double checking namespace here that this is where we want these tools? are we by any chance allowing this to be configurable?

File: fluentd-daemonset.yaml [hunk 2]
```diff
@@ -1,23 +1,77 @@
...
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: fluentd-daemon
+  labels:
+    component: fluentd-daemon
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
+rules:
+- apiGroups:
```
Comment (alejandroEsc) [inline fluentd-daemonset.yaml]: [*]  all apis?

File: fluentd-daemonset.yaml [hunk 3]
```diff
@@ -1,23 +1,77 @@
...
+  - ""
+  resources:
+  - "namespaces"
+  - "pods"
+  verbs:
```
Comment (alejandroEsc) [inline fluentd-daemonset.yaml]: ["get", "watch", "list"]

File: fluentd-daemonset.yaml [hunk 4]
```diff
@@ -1,23 +1,77 @@
...
+        component: fluentd-daemon
         version: v1
         kubernetes.io/cluster-service: "true"
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ''
     spec:
+      serviceAccountName: fluentd-daemon
       containers:
-      - name: fluentd-kafka
+      - name: fluentd-daemon
```
Comment (alejandroEsc) [inline fluentd-daemonset.yaml]: seems like you are doing a bit more than just rbac, details about some of the additional label and name changes?

Comment (leahnp) [inline fluentd-daemonset.yaml]: I categorized this as "clean up" in the PR - I just wanted to create some uniformity with the naming. If you feel like I should do a different PR for these changes, let me know.

File: fluentd-daemonset.yaml (hunk 1 above)
Comment (leahnp) [inline fluentd-daemonset.yaml]: It was in `default` but it seems like the k8s community leans toward putting logging in kube-system, I am open to suggestions.

File: fluentd-daemonset.yaml (hunk 3 above)
Comment (leahnp) [inline fluentd-daemonset.yaml]: I don't understand this comment -

Comment (leahnp) [inline fluentd-daemonset.yaml]: Oh - you want me to write the verbs horizontally and not list them? What is the difference?

File: fluentd-daemonset.yaml (hunk 1 above)
Comment (alejandroEsc) [inline fluentd-daemonset.yaml]: I see, got more context into this, if this is where it is usually put, thats cool. In the chart we should leave this out of course and let the user decide what namespace.

File: fluentd-daemonset.yaml (hunk 3 above)
Comment (alejandroEsc) [inline fluentd-daemonset.yaml]: yep please, its a bit clearer when comparing what verbs you are vs what resources you are allowed.

File: fluentd-daemonset.yaml (hunk 4 above)
Comment (alejandroEsc) [inline fluentd-daemonset.yaml]: Oh I see, the description mentions this. ok.

File: fluentd-daemonset.yaml (hunk 2 above)
Comment (alejandroEsc) [inline fluentd-daemonset.yaml]: by this comment I mean are you really asking for all apis?

Comment (leahnp) [inline fluentd-daemonset.yaml]: my understanding of using `""` is that it adds all the `core apis` opposed to `"*"` which would add ALL the apis. I saw both used in fluentd logging RBAC examples...I tried reading through the k8s API docs, but I'm afraid my understanding of which API's fall under `core` and which would encompass `all` is still a bit hazy - please enlighten me if you can, thanks!

File: fluentd-daemonset.yaml (hunk 3 above)
Comment (leahnp) [inline fluentd-daemonset.yaml]: ok - should I do APIs and resources horizontally also?

File: fluentd-daemonset.yaml (hunk 2 above)
Comment (alejandroEsc) [inline fluentd-daemonset.yaml]: Ok, so two things one: `[""]` horizontally. Yes you are right, you are looking for the core api, while "*" would be all. For some reason i thought i saw "*" so I apologize. You declare what resources from core api groups you want via the resource key there, so all good. I think and i'll have to review this, all would mean all available apis that are exposed, and so potentially any and all those resources that are not core resources also.

**Comment by leahnp** [comment]
@alejandroEsc changes made, let me know if that's what you were looking for - thanks for reviewing!

**Comment by alejandroEsc** [comment]
LGTM
